scripts/plasticc/plasticc.py

```diff
@@ -0,0 +1,171 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datasets
+from datasets import Features, Value, Array2D, Sequence
+from datasets.data_files import DataFilesPatternsDict
+from pathlib import Path
+import itertools
+import h5py
+import numpy as np
+
+# TODO: Add BibTeX citation
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+author={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+Simulated LSST astronomical time-series dataset from the PLAsTiCC Kaggle challenge
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = ""
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_VERSION = "0.0.1"
+
+_FLOAT_FEATURES = [
+        'hostgal_photoz',
+        'hostgal_specz',
+        'redshift',
+    ]
+
+_INT_FEATURES = [
+        'obj_type'
+    ]
+
+class PLAsTiCC(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = _VERSION
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="plasticc",
+            version=VERSION,
+            data_files=DataFilesPatternsDict.from_patterns({"train": ["healpix=*/train*.hdf5"], "test": ["healpix=*/test*.hdf5"]}),
+            description="train: plasticc train (spectroscopic), test: plasticc test (photometric)",
+        ),
+        datasets.BuilderConfig(name="train_only",
+                                version=VERSION,
+                                data_files=DataFilesPatternsDict.from_patterns({"train": ["healpix=*/train*.hdf5"]}),
+                                description="load train (spectroscopic) data only"),
+        datasets.BuilderConfig(name="test_only",
+                                version=VERSION,
+                                data_files=DataFilesPatternsDict.from_patterns({"train": ["healpix=*/test*.hdf5"]}),
+                                description="load test (photometric) data only"),
+    ]
+
+    DEFAULT_CONFIG_NAME = "train_only"
+
+    @classmethod
+    def _info(self):
+        """ Defines the features available in this dataset.
+        """
+        # Starting with all features common to time series datasets
+        features = {
+            'lightcurve': Sequence(feature={
+                'band': Value('int32'),
```

Comment:
Are the bands just in integer values? And not strings (e.g., ugrizY)? Should we add a mapping or description somewhere of that?

Reply:
@helenqu to fix this, I am storing the strings as an ordered sequence of strings matching the indicies of `bands`. :+1: See [this line](https://github.com/AstroPile/AstroPile_prototype/blob/90a1ef4e3c2dbc63c514b8bc8559e73666fc09b5/scripts/yse/yse.py#L167) (and related).